src/agent/agent_gather.rs

```diff
@@ -98,11 +105,17 @@ impl Agent {
                     });
                 }
                 CandidateType::ServerReflexive => {
+                    let ephemeral_config = match &params.udp_network {
```

Comment:
from pion/ice, I didn't see any special handling to disable server reflexive for UdpMux.

Reply:
Hmm. `pion/ice` does gather candidates using server reflexive when doing UDP muxing, however it doesn't respect the muxing config and instead uses ephemeral connections for this. 

For a server side use case when the network/firewall connection is known this feels wrong? In my opinion we should either disable server reflexive candidate gathering, as I have here, or perform it, but do so using the muxed connection not ephemeral connections.

I think what `pion/ice` does currently is not correct and violates the expectation of the user, namely that when doing UDP muxing no other udp sockets are created/used.